tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java

```diff
@@ -120,7 +121,10 @@ public <T> T getJaxbRoot(InputStream stream, Class<T> clazz, Schema schema)
 		ContextProvidingValidationEventHandler handler = new ContextProvidingValidationEventHandler();
 		try {
 			staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
-			JAXBContext jaxbContext = JAXBContext.newInstance( ObjectFactory.class );
+			ClassLoader cl = this.getClass().getClassLoader();
+			String packageName = Optional.ofNullable(ObjectFactory.class.getPackage())
+					.map(p -> p.getName()).orElse("org.hibernate.jpamodelgen.xml.jaxb");
```

Comment:
I don't understand what's going on here. Can `ObjectFactory` somehow change packages??

Why is this not just a constant?

Reply:
In fact, I get checkstyle warnings : since the `ObjectFactory.class.getPackage()` could return `null` (in fact it doesn't but ...)
The fix changes from class ref `JAXBContext.newInstance(ObjectFactory.class)` to package ref (since we need to pass classloader) `JAXBContext.newInstance( packageName, cl )`

Reply:
I've just forced push since there was a refactor in package names

Reply:
This is the kind of error if I change the code from above to the following :

```java
staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
ClassLoader cl = this.getClass().getClassLoader();
JAXBContext jaxbContext = cl == null ? JAXBContext.newInstance( ObjectFactory.class.getPackage().getName() ) : JAXBContext.newInstance( ObjectFactory.class.getPackage().getName(), cl );
Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
```

Error :
```
.../hibernate-orm/tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java:125: error: [dereference.of.nullable] dereference of possibly-null reference ObjectFactory.class.getPackage()
                        JAXBContext jaxbContext = cl == null ? JAXBContext.newInstance( ObjectFactory.class.getPackage().getName() ) : JAXBContext.newInstance( ObjectFactory.class.getPackage().getName(), cl );
                                                                                                                      ^
.../hibernate-orm/tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java:125: error: [dereference.of.nullable] dereference of possibly-null reference ObjectFactory.class.getPackage()
                        JAXBContext jaxbContext = cl == null ? JAXBContext.newInstance( ObjectFactory.class.getPackage().getName() ) : JAXBContext.newInstance( ObjectFactory.class.getPackage().getName(), cl );
                                                                                                                                                                                              ^
```

Reply:
Forced push with suggested change 😉

Reply:
Don't you want the same change for the `ClassLoader`?

I mean, if there's no class loader when we're expecting there to be one, then that's something we want to report, no?

Reply:
I dont know if you can have situations were it will not have classloader returned from the getter.
The constructor allow single argument with no classloader which delegates to the two args constructor with a self-handled classloader

Reply:
If you think it's a non-problem, I can for sure do the same change 👍

Reply:
But _that's exactly the point_. If you _don't know_, then your code potentially introduces weird arbitrary undefined behavior that might be almost impossible to troubleshoot.

We need things to _blow up_ when something "weird" happens, not just carry on doing something not well-defined!

Plus, when someone comes along later and reads that code, it's completely baffling, and they're scared to touch it, because they don't know what you were thinking of in the "null classloader" case.

My biggest gripe on earth is developers believing that errors are their enemy. The opposite is true: errors are your best friend as a developer.

Reply:
I don't want to introduce any regressions that's why I was careful (maybe too much) about the PR modifications introduced.
I've done the same thing for `classloader` now